shared/ui-components/src/helpers/useLanguageChange.ts

```diff
@@ -0,0 +1,35 @@
+import { useRouter } from "next/router"
+import { LangItem } from "../navigations/LanguageNav/LanguageNav"
+
+export function useLanguageChange(languages: LangItem[]) {
+  const router = useRouter()
+
+  function change(prefix: string): void {
+    if (typeof window == "undefined") return
+
+    const pathname = window.location.pathname
+
+    let newPath = pathname
+
+    const pathIncludesLang = languages.filter((item) => {
+      const pattern = new RegExp(`(/${item.prefix}/$)|(/${item.prefix}$)|(/${item.prefix}/)`, "gm")
```

Comment:
I think all you'd need to use here is something like this:

`^/(${item.prefix})(/|$)`

You either want to match `/es` (end of string) or `/es/` (plus whatever), and you'll have the prefix as a capture group in case you ever need it. You wouldn't want to match "/some/path/including/es/and/more"